pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/memory/EmptyIndexBuffer.java

```diff
@@ -0,0 +1,304 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.spi.memory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Properties;
+import org.roaringbitmap.buffer.ImmutableRoaringBitmap;
+
+
+/**
+ * A specialized PinotDataBuffer implementation for zero-size index entries that contains S3 segment path information.
+ * This buffer is useful for debugging and tracking purposes when dealing with empty index entries.
+ */
+public class EmptyIndexBuffer extends PinotDataBuffer {
+  private final Properties _properties;
+  private final String _segmentName;
+  private final String _tableNameWithType;
+  private final String _segmentPath;
+
+  /**
+   * Creates a new S3EmptyIndexBuffer for a zero-size index entry
+   *
+   * @param properties Properties containing S3 configuration (bucket, key, etc.)
+   * @param segmentName The name of the segment
+   * @param tableNameWithType The table name with type
+   */
+  public EmptyIndexBuffer(Properties properties, String segmentName, String tableNameWithType) {
+    super(false); // Not closeable since it's just metadata
+    _properties = properties;
+    _segmentName = segmentName;
+    _tableNameWithType = tableNameWithType;
+    _segmentPath = constructSegmentPath();
+  }
+
+  /**
+   * Constructs the S3 segment path from properties
+   * @return The constructed S3 URI as a string
+   */
+  private String constructSegmentPath() {
```

Comment:
make it abstract and have another `S3EmptyIndexBuffer` extends this `EmptyIndexBuffer`?